datafusion/physical-expr-common/src/physical_expr.rs

```diff
@@ -283,6 +284,47 @@ pub trait PhysicalExpr: Send + Sync + Display + Debug + DynEq + DynHash {
     /// See the [`fmt_sql`] function for an example of printing `PhysicalExpr`s as SQL.
     ///
     fn fmt_sql(&self, f: &mut Formatter<'_>) -> fmt::Result;
+
+    /// Take a snapshot of this `PhysicalExpr` if it is dynamic.
+    /// This is used to capture the current state of `PhysicalExpr`s that may contain
+    /// dynamic references to other operators in order to serialize it over the wire
+    /// or treat it via downcast matching.
+    ///
+    /// You should not call this method directly as it does not handle recursion.
+    /// Instead use `shapshot_physical_expr` to handle recursion and capture the
```

Comment:
I think if you do it like this:

```suggestion
    /// Instead use [`shapshot_physical_expr`] to handle recursion and capture the
```

rustdoc will add a link automatically for you